client/src/components/JurisdictionAdmin/BatchInventory.tsx

```diff
@@ -31,43 +31,119 @@ import {
   stepState,
   StepPanelColumn,
 } from '../Atoms/Steps'
-import { Column } from '../Atoms/Layout'
+import { Column, Row } from '../Atoms/Layout'
+import {
+  BatchInventoryConfig,
+  useBatchInventoryFeatureFlag,
+} from '../useFeatureFlag'
+
+type Step =
+  | 'Select System Type'
+  | 'Upload Election Results'
+  | 'Inventory Batches'
+  | 'Download Audit Files'
 
-const STEPS = [
+const STEPS: Step[] = [
+  'Select System Type',
   'Upload Election Results',
   'Inventory Batches',
   'Download Audit Files',
-] as const
+]
+
+const STEPS_IF_NOT_SHOWING_BALLOT_MANIFEST: Step[] = [
+  'Select System Type',
+  'Upload Election Results',
+  'Download Audit Files',
+]
 
 const isUploaded = (fileUpload: IFileUpload) =>
   fileUpload.uploadedFile.data?.processing?.status ===
   FileProcessingStatus.PROCESSED
 
+function areAllFilesUploaded({
+  systemType,
+  cvrUpload,
+  tabulatorStatusUpload,
+}: {
+  systemType: CvrFileType
+  cvrUpload: IFileUpload
+  tabulatorStatusUpload: IFileUpload
+}): boolean {
+  if (systemType === CvrFileType.DOMINION) {
+    return isUploaded(cvrUpload) && isUploaded(tabulatorStatusUpload)
+  }
+  return isUploaded(cvrUpload)
+}
+
+function batchInventoryQueryKey(
+  jurisdictionId: string,
+  dataType: 'systemType' | 'cvr' | 'tabulatorStatus' | 'signOff'
+): string[] {
+  return ['batchInventory', jurisdictionId, dataType]
+}
+
+interface ISystemTypeQueries {
+  systemType: UseQueryResult<CvrFileType | null>
+  setSystemType: UseMutationResult<CvrFileType, unknown, CvrFileType, unknown>
+}
+
+const useBatchInventorySystemType = (
+  electionId: string,
+  jurisdictionId: string
+): ISystemTypeQueries => {
+  const key = batchInventoryQueryKey(jurisdictionId, 'systemType')
+  const url = `/api/election/${electionId}/jurisdiction/${jurisdictionId}/batch-inventory/system-type`
+  const queryClient = useQueryClient()
+  return {
+    systemType: useQuery<CvrFileType | null>(
+      key,
+      async () => (await fetchApi(url)).systemType
+    ),
+    setSystemType: useMutation(
+      (systemType: CvrFileType) =>
+        fetchApi(url, {
+          method: 'PUT',
+          headers: { 'Content-Type': 'application/json' },
+          body: JSON.stringify({ systemType }),
+        }),
+      {
+        onSuccess: () =>
+          Promise.all(
+            ([
+              'systemType',
+              'cvr',
+              'tabulatorStatus',
+              'signOff',
+            ] as const).map(dataType =>
```

Comment:
Changing the system type after uploading files requires re-uploading files. You'll see in the next commit that the backend does in fact clear everything on system type change.

Can add a comment noting this. Dropped some related comments in this file by mistake and will add those back too